tests/integration/test_sql_init.py

```
@@ -0,0 +1,109 @@
+        )
+        assert is_nullable == "NO", f"raw.{table}.ingested_at must remain NOT NULL"
+
+
+@pytest.mark.parametrize("table", RAW_TABLES)
+def test_raw_insert_with_defaults_round_trip(pg_conn, table: str) -> None:
+    """Insert supplying only the non-defaulted columns and verify defaults fire."""
+    with pg_conn.cursor() as cur:
+        cur.execute(
+            f"INSERT INTO raw.{table} (event_type, source, payload) "
+            "VALUES (%s, %s, %s::jsonb) "
+            "RETURNING event_id, ingested_at",
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)